# FluentUI components should not be empty (`no-empty-components-v9`)

Please describe the origin of the rule here.

## Rule Details

This rule aims to...

Examples of **incorrect** code for this rule:

```jsx
<Text></Text>
```

```jsx
<Label></Label>
```

Examples of **correct** code for this rule:

```jsx
<Label size="small">Small</Label>
```

```jsx
<Label size="small">Small</Label>
```

```jsx
<Text>This is an example of the Text component's usage.</Text>
```

### Options

If there are any options, describe them here. Otherwise, delete this section.

## When Not To Use It

Give a short description of when it would be appropriate to turn off this rule.

## Further Reading

If there are other links that describe the issue this rule addresses, please include them here in a bulleted list.

